src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -531,13 +531,19 @@ def get_dmatrix(data_path, content_type, csv_weights=0, is_pipe=False):
             shutil.rmtree(files_path, ignore_errors=True)
             os.mkdir(files_path)
             for path in data_path:
+                # logging for debugging remove before final merge
+                print(path)
                 if not os.path.exists(path):
                     return None
-                if os.path.isfile(path):
-                    os.symlink(path, os.path.join(files_path, os.path.basename(path)))
+                symlink_dest = os.path.join(files_path, os.path.basename(path))
+                if os.path.isfile(path) and not os.path.exists(symlink_dest):
+                    os.symlink(path, symlink_dest)
                 else:
+
                     for file in os.scandir(path):
-                        os.symlink(file, os.path.join(files_path, file.name))
+                        symlink_dest_dir = os.path.join(files_path, file.name)
```

Comment:
we need to test here how using 1 file instead of 2 when things collide if this will cause different expected results, and either emit warning or make the names different so that they don't collide. If we make new names, we need to make sure those new file names are used somehow